### PR TITLE
Tila 1528: Adds NotificationList component to common, implements it on ui-side

### PR DESCRIPTION
**Pull request by vincit-matu**

### CODE REVIEW

File: common/src/components/NotificationsList.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  const displayedNotificationsList = groupedNotificationsList?.filter(
+    (item) => {
+      if (
+        !closedNotificationsList ||
+        closedNotificationsList.includes(String(item?.id)) ||
+        (item?.target !== target && item?.target !== "ALL")
+      ) {
+        return false;
+      }
+      return true;
```
Comment (joonatank) [inline common/src/components/NotificationsList.tsx:147]: `return (!closedNotif...)`? or with just: `(item => (!closedNotif...)`

File: common/src/components/NotificationsList.tsx [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  let notificationType;
+  switch (notification.level) {
+    case "EXCEPTION":
+      notificationType = "alert" as NotificationType;
+      break;
+    case "WARNING":
+      notificationType = "error" as NotificationType;
+      break;
+    default:
+      notificationType = "info" as NotificationType;
```
Comment (joonatank) [inline common/src/components/NotificationsList.tsx:82]: Does `let notificationType: NotificationType = 'info'` or `notificationType = 'error' as const` remove the need for `as` cast?

**Review by joonatank — APPROVED** [review]

File: common/src/components/NotificationsList.tsx (hunk 1 above)
Comment (vincit-matu) [reply common/src/components/NotificationsList.tsx:147]: Haha! *facepalm* Indeed.. this became the much more concise:
`(item) =>
      !(closedNotificationsList as string[]).includes(String(item?.id)) ||
      !(item?.target !== target && item?.target !== "ALL")`

File: common/src/components/NotificationsList.tsx (hunk 2 above)
Comment (vincit-matu) [reply common/src/components/NotificationsList.tsx:82]: Yep, indeed it does. Much cleaner to declare the type than to cast the variables - cheers!

**Pull request merged**